extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaBindingConverter.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.kafka.client.runtime;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+public class KafkaBindingConverter implements ServiceBindingConverter {
+
+    @Override
+    public Optional<ServiceBindingConfigSource> convert(List<ServiceBinding> serviceBindings) {
+        Optional<ServiceBinding> matchingByType = ServiceBinding.singleMatchingByType("kafka", serviceBindings);
+        if (!matchingByType.isPresent()) {
+            return Optional.empty();
+        }
+
+        Map<String, String> properties = new HashMap<>();
+        ServiceBinding binding = matchingByType.get();
+
+        String bootstrapServers = binding.getProperties().get("bootstrapServers");
+        if (bootstrapServers == null) {
+            bootstrapServers = binding.getProperties().get("bootstrap-servers");
+        }
+        if (bootstrapServers != null) {
+            properties.put("kafka.bootstrap.servers", bootstrapServers);
+        }
+
+        String securityProtocol = binding.getProperties().get("securityprotocol");
+        if (securityProtocol != null) {
+            properties.put("kafka.security.protocol", securityProtocol);
+        }
+
+        String saslMechanism = binding.getProperties().get("saslmechanism");
+        if (saslMechanism != null) {
+            properties.put("kafka.sasl.mechanism", saslMechanism);
+        }
+        String user = binding.getProperties().get("user");
+        String password = binding.getProperties().get("password");
+        if ("PLAIN".equals(saslMechanism) && (user != null) && (password != null)) {
+            properties.put("kafka.sasl.jaas.config",
+                    String.format("org.apache.kafka.common.security.plain.PlainLoginModule required %s %s", user, password));
```

Comment:
Here it's tricky. Format and module implementation are deployment-specific. By doing this, we prevent the usage of service bindings when the kafka broker use Kerberos or another type of authentication.

Reply:
I believe it's fine for now, but we may need to add a hint (a file) in the service bindings.

Reply:
Yeah, this definitely need to be improved when we have more supported types